frontend/public/components/secrets/create-secret.tsx

```diff
@@ -121,174 +130,181 @@ const determineSecretType = (stringData): SecretType => {
+                  <p className="help-block" id="secret-name-help">
+                    {t('secret~Unique name of the new secret.')}
+                  </p>
+                </div>
               </div>
-            </div>
-          </fieldset>
-          <SubForm
-            onChange={this.onDataChanged}
-            onError={this.onError}
-            onFormDisable={this.onFormDisable}
-            stringData={this.state.stringData}
-            secretType={this.state.secret.type}
-            isCreate={this.props.isCreate}
-          />
-        </>
-      );
-    };
+            </fieldset>
+            <SubForm
+              onChange={this.onDataChanged}
+              onError={this.onError}
+              onFormDisable={this.onFormDisable}
+              stringData={this.state.stringData}
+              secretType={this.state.secret.type}
+              isCreate={this.props.isCreate}
+            />
+          </>
+        );
+      };
 
-    render() {
-      const { secretTypeAbstraction } = this.state;
-      const { onCancel = history.goBack } = this.props;
-      const title = `${this.props.titleVerb} ${secretDisplayType(secretTypeAbstraction)} Secret`;
-      return modal ? (
-        <form className="co-create-secret-form modal-content" onSubmit={this.save}>
-          <ModalTitle>{title}</ModalTitle>
-          <ModalBody>{this.renderBody()}</ModalBody>
-          <ModalSubmitFooter
-            errorMessage={this.state.error || ''}
-            inProgress={this.state.inProgress}
-            submitText="Create"
-            cancel={this.props.onCancel}
-          />
-        </form>
-      ) : (
-        <div className="co-m-pane__body">
-          <Helmet>
-            <title>{title}</title>
-          </Helmet>
-          <form
-            className="co-m-pane__body-group co-create-secret-form co-m-pane__form"
-            onSubmit={this.save}
-          >
-            <h1 className="co-m-pane__heading">{title}</h1>
-            <p className="co-m-pane__explanation">{this.props.explanation}</p>
-            {this.renderBody()}
-            <ButtonBar errorMessage={this.state.error} inProgress={this.state.inProgress}>
-              <ActionGroup className="pf-c-form">
-                <Button
-                  type="submit"
-                  data-test="save-changes"
-                  isDisabled={this.state.disableForm}
-                  variant="primary"
-                  id="save-changes"
-                >
-                  {this.props.saveButtonText || 'Create'}
-                </Button>
-                <Button type="button" variant="secondary" id="cancel" onClick={onCancel}>
-                  Cancel
-                </Button>
-              </ActionGroup>
-            </ButtonBar>
+      render() {
+        const { secretTypeAbstraction } = this.state;
+        const { onCancel = history.goBack } = this.props;
+        const { t } = this.props;
+        const title = t('secret~{{verb}} {{type}} Secret', {
```

Comment:
We probably need separate messages for `Create {{type}} Secret` and `Edit {{type}} Secret` instead of concatenating strings.

Reply:
Hi @spadgett 
The verb will be passed in as a translated string by `titleVerb` prop (https://github.com/pallypan/console/blob/master/frontend/public/components/secrets/create-secret.tsx#L1292), it will be hard to separate message as we do not know we should use Create or Edit.
The way I think of is change `titleVerb` prop to a hardcode string instead of a translated one, so we can check its value and determine which string to use. But `withSecretForm` is an exported component and used by https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/import/CreateSecretModal.tsx#L3 --- I'm nervous to change `titleVerb` prop defination